### PR TITLE
Add exercise: run-length-encoding

### PR DESCRIPTION
**Pull request by Teo-ShaoWei**
Another call-to-arm to #87.

### CODE REVIEW

File: exercises/run-length-encoding/example.jl [hunk 1]
```diff
@@ -0,0 +1,26 @@
+function encode(s)
+    (length(s) == 0) && (return s)
+
+    count = 0
+    for char in s
+        if char == s[1]
```
Comment (SaschaMann) [inline exercises/run-length-encoding/example.jl]: This could also use the ternary operator.

**Comment by SaschaMann** [comment]
This would fit fairly well in the "count letters or words"-section of the progression to be unlocked by nucleotide-count, I think. Your thoughts on that?

File: exercises/run-length-encoding/example.jl (hunk 1 above)
Comment (Teo-ShaoWei) [reply exercises/run-length-encoding/example.jl]: Wow thanks for being detailed with the check! =)

This one I intended for the `break` to be visible and thus settled on this structure, but I was perplexed as I spent quite a while thinking about which is better, without a clear winner for me in the end. Ternary with control flow like `break` will make the code terse and appeal to stronger coders but might be off-putting to beginners who will find it too hard to read, kinda like how Haskell is...

I think I can respect your opinion on this. Will edit.

**Comment by Teo-ShaoWei** [comment]
Yes, I agree with you that this is of a similar nature. Let's park it that way. Sorry but how should I correctly implement that? Is it in `config.json`?

**Comment by SaschaMann** [comment]
Yes. You just set `"unlocked_by"` to the exercise slug that unlocks it. You can check if it worked with `confliget tree .`.

File: exercises/run-length-encoding/example.jl (hunk 1 above)
Comment (SaschaMann) [reply exercises/run-length-encoding/example.jl]: Imo the syntax highlighter will make it visible enough but I see your point. In the end it's mostly personal preference.

**Comment by Teo-ShaoWei** [comment]
Ok done as suggested.

**Pull request merged**